src/Dialog/Dialog.tsx

```diff
@@ -387,17 +382,16 @@ const Buttons: React.FC<React.PropsWithChildren<{buttons: DialogButtonProps[]}>>
   return (
     <>
       {buttons.map((dialogButtonProps, index) => {
-        const {content, buttonType = 'normal', autoFocus = false, ...buttonProps} = dialogButtonProps
-        const ButtonElement = buttonTypes[buttonType]
+        const {children, buttonType = 'default', autoFocus = false, ...buttonProps} = dialogButtonProps
```

Comment:
I also think maybe we shouldn't rename `content` to `children`. Because `content` here is more like a [button text content](https://github.com/primer/react/blob/main/src/stories/Dialog.stories.tsx#L192) rather than a typical [React children prop](https://react.dev/learn/passing-props-to-a-component#passing-jsx-as-children)

Reply:
I am mainly concerned because a prop rename will potentially cause breaking changes and also Dialog props are a combined with `ButtonProps` so it would end up having two `children` props. https://github.com/primer/react/blob/main/src/Dialog/Dialog.tsx#L23

Same concern for renaming the `buttonType` option here as well https://github.com/primer/react/pull/3084/files#diff-7c9e4f2540e063c93f543e6eaa44cac7dc3b0e55a8fe81030c9e571f72950a3eR27

Would that be possible to revert these as well? And we can chat about them further if you are interested 😊

Reply:
> Same concern for renaming the buttonType option here as well https://github.com/primer/react/pull/3084/files#diff-7c9e4f2540e063c93f543e6eaa44cac7dc3b0e55a8fe81030c9e571f72950a3eR27

I might be looking at it wrong but I believe that this rename for the `buttonType` is compliant with the non-deprecated `Button` ( https://github.com/primer/react/blob/main/src/Button/types.ts#L11 )

Regarding the `children` vs `content`, should we then `Omit` `children` on this extension of `Button`? https://github.com/primer/react/pull/3084/files#diff-7c9e4f2540e063c93f543e6eaa44cac7dc3b0e55a8fe81030c9e571f72950a3eR23

Since we're using `content` for the prop on `DialogButtons` but `Button` has a `children`, there's a slight mismatch, at the moment, since, for example, the `footerButtons` on the `stories`, require a `content` and a `children` property.

Reply:
I wouldn't mind supporting `children` but definitely wanted to +1 keeping `content` available during this transition so we don't break any existing usage using `content`. Do either of you think this could be possible? Something like: `{children ?? content}` if we end up using both here, and keep `content` on the `DialogButtonProps`.

Just an idea, let me know what you think!

Reply:
Thanks @joshblack! I love that suggestion! I played with the code a bit today to see how we can support both but the typescript is making trouble and needs `children` because `DialogButtonProps` has `children` as a mandatory prop through `Buttons`. Is there anything I am missing here? I would really love to support both in the transition period! Is there a way to make `Button` 's mandatory `children` prop optional in `DialogButtonProps`? 😬  Or any other way really.


@ItzaMi you are 💯 right on renaming the `buttonType` from being `normal` to `default` (sorry I missed that earlier) however this is still a breaking change and we really would love to merge this before the major release so following @joshblack's suggestion for supporting both, how about we add `default` as another option?

```
/**
   * The type of Button element to use
   */
  buttonType?: 'normal' | 'primary' | 'danger' | 'default'
```

And on the `<Button>` element, we could make sure we pass `default` even if we receive `normal`

```
<Button
....
variant={buttonType === 'normal' ? 'default' : buttonType}
...
>
```

Looking forward to hearing your thoughts! 🙌🏻